### PR TITLE
[MAD][DATA1018][lab-api-scavenger-game] Ivan Cernicharo

### PR DESCRIPTION
**Pull request by ijcernicharo**
Hello Belen,
How are you? Hope fine and resting right now.
I have almost finished this lab. Many difficulties found, like how the hell make a request of the whole repository in recursive mode listing all modules and files (I have achieved it finnally :sweat_smile: )
I am finishing the read iteration for hidden files to get the secret message.
Thanks again for your feedback.
Best regards,
Ivan

### CODE REVIEW

**Comment by ijcernicharo** [comment]
Hello again Belen,
I have just finished the lab. It was driving me crazy by how to find a method to read hidden files.... and the solution was to read them as a simple doc with the typical open(filename,'r'). Finished and obtained the cold joke. In my opinion, I think complaining time is almost the 50% of the time :sweat_smile: .
I look forward to receiving your feedback of this lab.
Best regards,
Ivan

**Comment by ijcernicharo** [comment]
Hello Belen,
Sorry for the last update. As Marc said, I have developed a way to obtain the cold joke via urls instead of obtaining it from reading local hidden files. Seems inefficient to me, more when we have a limit of 60 request in an hour. By the way, right now We have excedeed the limit for local Ionhack Students network. You will see my code recognizes this problem and print a message to warn you about it.
Thanks again for your time.
Best regards,
Ivan

**Comment by ijcernicharo** [comment]
P.S.: last modification for challenge 3 is the one called lab-api-scavenger-game MOD#3 Finished hardcore path for challlenge 3

File: module-1/lab-api-scavenger-game/your-code/challenge_1.ipynb [hunk 1]
```diff
@@ -0,0 +1,197 @@
+{
```
Comment (Gobinde43) [inline module-1/lab-api-scavenger-game/your-code/challenge_1.ipynb]: Good job Ivan!! Well done!! You can do it in less code lines. Normalizing the JSON, creating a dataframe and getting the unique values of a dataframe column.

File: module-1/lab-api-scavenger-game/your-code/challenge_2.ipynb [hunk 2]
```diff
@@ -0,0 +1,651 @@
+{
```
Comment (Gobinde43) [inline module-1/lab-api-scavenger-game/your-code/challenge_2.ipynb]: You have solved Q1 in this one like a PRO. More readable code!! Congrats!!

File: module-1/lab-api-scavenger-game/your-code/challenge_3.ipynb [hunk 3]
```diff
@@ -26,7 +26,8 @@
     "import json\n",
```
Comment (Gobinde43) [inline module-1/lab-api-scavenger-game/your-code/challenge_3.ipynb]: Always the second approach!! Great code the hardcore path. Congrats!!
Remember to keep your for loops under control!!! jejejej

**Pull request closed without merging**